packages/web-ui/src/sections/Document/Sidebar/Files/index.tsx

```diff
@@ -276,41 +277,100 @@ function FileNode({
   createFile,
   destroyFile,
+  destroyFolder,
+  isDestroying,
 }: {
   createFile: (args: { path: string }) => Promise<void>
   destroyFile: (documentUuid: string) => Promise<void>
+  destroyFolder: (path: string) => Promise<void>
   documents: SidebarDocument[]
   currentPath: string | undefined
   navigateToDocument: (documentUuid: string) => void
+  isDestroying: boolean
 }) {
   const togglePath = useOpenPaths((state) => state.togglePath)
   const rootNode = useTree({ documents })
+  const [deletableNode, setDeletable] =
+    useState<DeletableElement<DeletableType> | null>(null)
 
   useEffect(() => {
     if (currentPath) {
       togglePath(currentPath)
     }
   }, [currentPath, togglePath])
+  const onConfirmDelete = useCallback(
+    async <T extends DeletableType>(deletable: DeletableElement<T>) => {
+      if (deletable.type === 'file') {
+        await destroyFile(deletable.documentUuid)
+      } else if (deletable.type === 'folder') {
+        await destroyFolder(deletable.path)
+      }
 
+      setDeletable(null)
+    },
+    [destroyFile, destroyFolder, deletableNode, setDeletable],
+  )
+
+  const deletingFolder = deletableNode?.type === 'folder'
   return (
-    <FileTreeProvider
-      currentPath={currentPath}
-      onNavigateToDocument={navigateToDocument}
-      onCreateFile={async (path) => {
-        createFile({ path })
-      }}
-      onDeleteFile={destroyFile}
-      onDeleteFolder={async (path) => {
-        console.log('onDeleteFolder', path)
-      }}
-    >
-      <FileNode node={rootNode} />
-    </FileTreeProvider>
+    <>
+      <FileTreeProvider
+        currentPath={currentPath}
+        onNavigateToDocument={navigateToDocument}
+        onCreateFile={async (path) => {
+          createFile({ path })
+        }}
+        onDeleteFile={({ node, documentUuid }) => {
+          setDeletable({ type: 'file', documentUuid, name: node.name })
+        }}
+        onDeleteFolder={async ({ node, path }) => {
+          setDeletable({ type: 'folder', path, name: node.name })
+        }}
+      >
+        <FileNode node={rootNode} />
+      </FileTreeProvider>
+      {deletableNode ? (
+        <ConfirmModal
+          type='destructive'
+          open
+          onConfirm={() => onConfirmDelete(deletableNode)}
+          onOpenChange={(open) => {
+            if (!open) {
+              setDeletable(null)
+            }
+          }}
+          title='Are you sure?'
+          description={
+            deletingFolder
+              ? `You're deleting ${deletableNode?.name} folder`
+              : `You're deleting ${deletableNode?.name} prompt`
+          }
+          confirm={{
+            isConfirming: isDestroying,
+            label: deletingFolder ? 'Delete folder' : 'Delete prompt',
+            description: `Deleting this ${deletingFolder ? 'folder' : 'prompt'} will also delete all the prompts it contains. This action cannot be undone.`,
+          }}
+        />
+      ) : null}
+    </>
```

Comment:
huge component consider splitting un smaller chunks

Reply:
Is ultra componentized already with hooks and wrappers